### ISSUE TITLE
508-defect-2 [SCREENREADER] Undefined in aria-label

### ISSUE THREAD

**Issue by Mottie** [department-of-veterans-affairs/va.gov-team#12955]
# 508-defect-2

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

Edit buttons in the ArrayField and ObjectField will have an `aria-label` set as `Edit undefined` when the `ui:title` and `schema.title` are undefined. It will also be set to `Edit  ` when the `ui:title` is set to a string with one space (a common practice).

Related: https://github.com/department-of-veterans-affairs/va.gov-team/issues/12821

## Acceptance Criteria

As a screenreader user, the edit button should describe the content that it edits.

## Environment

* Operating System: All
* Browser: All
* Screenreading device: All
* Server: All

## Steps to Recreate

1. Log in as user 228
2. Make your way to the POW page of form 526 - staging.va.gov/disability/file-disability-claim-form-21-526ez/pow
3. Add more than one Period of confinement
4. Inspect the "Edit" button aria-label

## Possible Fixes (optional)

- Add value from `ui:options.ariaLabelForEditButtonOnReview` (missing in `ArrayField`, but exists in `ObjectField`
- Fall back to `ui:title` >  `ui:options.reviewTitle` > schema title

## Screenshots or Trace Logs

<details><summary>Screenshot of problem</summary>

![](https://user-images.githubusercontent.com/136959/91224290-8287e780-e6e7-11ea-8d02-479a210b8e8c.png)
</details>

## Definition of done

- [ ] Set a proper edit button aria-label (more than just "Edit", and not "Edit undefined")

**Comment by vhenry07** [comment]
Design system team is taking care of this work.
